ext/otel_observer.c

```diff
@@ -432,16 +582,22 @@ static void observer_begin(zend_execute_data *execute_data, zend_llist *hooks) {
         return;
     }
 
-    zval params[6];
-    uint32_t param_count = 6;
+    zval params[8];
+    uint32_t param_count = 8;
+    HashTable *attributes;
+    ALLOC_HASHTABLE(attributes);
+    zend_hash_init(attributes, 0, NULL, ZVAL_PTR_DTOR, 0);
 
     func_get_this_or_called_scope(&params[0], execute_data);
-    func_get_args(&params[1], execute_data);
+    func_get_attribute_args(&params[6], attributes, execute_data);
+    func_get_args(&params[1], attributes, execute_data, true);
```

Comment:
If I'm reading this right, this function seems to collect attributes even if `WithSpan` was not present. Is there any point populating this if `WithSpan` was not used? If not, skipping that would save some time on collecting information that will never be used.

Reply:
Yes, that's true. I think that 4th param should consider if feature is enabled && WithSpan is used here...

Reply:
updated to only collect attribute if feature enabled and WithSpan exists on the function (or an interface) - this lookup could be another instance where keeping track of which function have/haven't got WithSpan...